code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -525,6 +525,15 @@
 		desc = "Your standard drinking glass."
 		return
 
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/pen))
+		var/txt = stripped_input(user, "What would you like to name this drink?", "Mixology", "", 30)
+		if(txt)
```

Comment:
This is a glass of _george melons is a traitor my body is in bar maint clone me pls_.